### ISSUE TITLE
I get problem when i convert checkpoint model to openvino

### ISSUE THREAD

**Issue by JinXiaozhao** [openvinotoolkit/openvino#9452]
##### System information (version)

I'm trying to convert frozen model or saved model to IR model.
**my model from** https://github.com/Adlik/Adlik/blob/master/benchmark/tests/test_model/resnet50_tensorflow/resnet50_ckpt.py
 Is there another way to do it other that writing extension to model optimizer?

Detailed description
root@7ac88c9dd8c2:~# python /opt/intel/openvino_2021.1.110/deployment_tools/model_optimizer/mo_tf.py --input_meta_graph /root/Adlik/benchmark/tests/test_model/resnet50_tensorflow/model/resnet50.ckpt.meta --output_dir  /root/compiler/mnist/ --input input_1 --output predictions/Softmax -b 1 --disable_nhwc_to_nchw 
Model Optimizer arguments:
Common parameters:
        - Path to the Input Model:      None
        - Path for generated IR:        /root/compiler/mnist/
        - IR output name:       resnet50.ckpt
        - Log level:    ERROR
        - Batch:        1
        - Input layers:         input_1
        - Output layers:        predictions/Softmax
        - Input shapes:         Not specified, inherited from the model
        - Mean values:  Not specified
        - Scale values:         Not specified
        - Scale factor:         Not specified
        - Precision of IR:      FP32
        - Enable fusing:        True
        - Enable grouped convolutions fusing:   True
        - Move mean values to preprocess section:       None
        - Reverse input channels:       False
TensorFlow specific parameters:
        - Input model in text protobuf format:  False
        - Path to model dump for TensorBoard:   None
        - List of shared libraries with TensorFlow custom layers implementation:        None
        - Update the configuration file with input/output node names:   None
        - Use configuration file used to generate the model with Object Detection API:  None
        - Use the config file:  None
Model Optimizer version:        2021.1.0-1237-bece22ac675-releases/2021/1
2021-12-27 12:33:37.349438: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2021-12-27 12:33:37.349524: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2021-12-27 12:33:39.517724: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory
2021-12-27 12:33:39.517777: W tensorflow/stream_executor/cuda/cuda_driver.cc:269] failed call to cuInit: UNKNOWN ERROR (303)
2021-12-27 12:33:39.517815: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (7ac88c9dd8c2): /proc/driver/nvidia/version does not exist
2021-12-27 12:33:39.518325: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
[ ERROR ]  Cannot infer shapes or values for node "conv1_bn/cond".
[ ERROR ]  Op type not registered 'conv1_bn_cond_true_frozen_5419' in binary running on 7ac88c9dd8c2. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed. while building NodeDef 'conv1_bn/cond/then/_0'
[ ERROR ]  
[ ERROR ]  It can happen due to bug in custom shape infer function <function tf_native_tf_node_infer at 0x7efbfcef3268>.
[ ERROR ]  Or because the node inputs have incorrect values/shapes.
[ ERROR ]  Or because input shapes are incorrect (embedded to the model or passed via --input_shape).
[ ERROR ]  Run Model Optimizer with --log_level=DEBUG for more information.
[ ERROR ]  Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.middle.PartialInfer.PartialInfer'>): Stopped shape/value propagation at "conv1_bn/cond" node. 
 For more information please refer to Model Optimizer FAQ, question #38. (https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html?question=38#question-38)

**Comment by Iffa-Meah** [comment]
Hi,

I noticed that your model format that is fed to the OpenVINO Model Optimizer is **resnet50.ckpt.meta**.
Generally, I believe you are trying to store non-frozen TensorFlow models and load them to the Model Optimizer.

If you intend to use **MetaGraph conversion (.meta file)**, these three or four files are required, and must be stored in the same directory:

1. model_name.meta
2. model_name.index
3. model_name.data-00000-of-00001 (digit part may vary)
4. checkpoint (optional)

Conversion command: python3 mo_tf.py --input_meta_graph <INPUT_META_GRAPH>.meta --output_dir <OUTPUT_MODEL_DIR>


Meanwhile, the Checkpoint format conversion required these two files:
1. inference_graph.pb or inference_graph.pbtxt
2. checkpoint_file.ckpt

Conversion command: python3 mo_tf.py --input_model <INFERENCE_GRAPH>.pb --input_checkpoint <INPUT_CHECKPOINT> --output_dir <OUTPUT_MODEL_DIR>

If you do not have an inference graph file, refer to [Freezing Custom Models in Python](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_TensorFlow.html#freeze-the-tensorflow-model).

You may carefully review and follow the steps provided in this [Loading Non-Frozen Models to the Model Optimizer](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_TensorFlow.html#loading-non-frozen-models-to-the-model-optimizer).

**Comment by JinXiaozhao** [comment]
Thanks for your reply.
These  four files all  be stored in the same director, but it can't.

root@7ac88c9dd8c2:~/Adlik/benchmark/tests/test_model/resnet50_tensorflow/model# tree
.
├── checkpoint
├── resnet50.ckpt.data-00000-of-00001
├── resnet50.ckpt.index
└── resnet50.ckpt.meta

0 directories, 4 files
root@7ac88c9dd8c2:~/Adlik/benchmark/tests/test_model/resnet50_tensorflow/model# python /opt/intel/openvino_2021.1.110/deployment_tools/model_optimizer/mo_tf.py --input_meta_graph /root/Adlik/benchmark/tests/test_model/resnet50_tensorflow/model/resnet50.ckpt.meta --output_dir  /root/compiler/mnist/

**Comment by Iffa-Meah** [comment]
I generated the script from the link that you gave and it produced 4 files:

![1_metafiles_from_script](https://user-images.githubusercontent.com/79643476/147556039-ec49bb65-6438-4d4f-82f6-5560447d14e3.png)

I tested your model with two conversion Methods:

**1.  Checkpoint**

The freezing script from [here](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_TensorFlow.html#freezing-custom-models-in-python) is used to get the inference_graph.pb file. 

![2_freeze](https://user-images.githubusercontent.com/79643476/147556250-5f4bf68b-3357-4b9f-a176-882593f5051e.png)

Since the graph file had been obtained, I took this chance to review your model's input and output shapes.
![3_NetronInput_shape](https://user-images.githubusercontent.com/79643476/147556365-2a195816-837a-438d-bfbb-d03061ebee8a.png)
![4_NetronOutput_shape](https://user-images.githubusercontent.com/79643476/147556389-fec60329-1385-4c18-8eec-0dc446bf9a35.png)

The conversion command that I used: 
python3 mo_tf.py --input_model /home/osboxes/Desktop/9452/model_FROZEN/inference_graph.pb --output_dir /home/osboxes/Desktop/9452/model_FROZEN/out --model_name resnet50 --batch 1

**Note: I tried with both mo.py and mo_tf.py and the results are the same**

![5_CheckpointConversion](https://user-images.githubusercontent.com/79643476/147556585-15b6bc3a-4dc5-47ed-9595-59dd8db318be.png)

However, this wasn't successful. I encountered the same error as yours: [ ERROR ]  Cannot infer shapes or values for node "conv1_bn/cond".

![6_CheckpointConversionError](https://user-images.githubusercontent.com/79643476/147556792-d40b2f4c-b4eb-4358-ae01-fa3fd630e311.png)


**2. MetaGraph**
This the command I used initially:
python3 mo.py --input_meta_graph /home/osboxes/Desktop/9452/model/resnet50.ckpt.meta --output_dir /home/osboxes/Desktop/9452/model/out/ 

However, there was a prompt about negative input shape.

![7_MetaGraphCnversion](https://user-images.githubusercontent.com/79643476/147556995-fa7f0280-006a-4acd-a99a-62733e5a4440.png)

So, I changed the command to:
python3 mo.py --input_meta_graph /home/osboxes/Desktop/9452/model/resnet50.ckpt.meta --output_dir /home/osboxes/Desktop/9452/model/out/ --input input_1 --input_shape [1,224,224,3] 

The conversion returns the same error unfortunately: 
[ ERROR ]  Cannot infer shapes or values for node "conv1_bn/cond".

![8_MetaGraphCnversion_Error](https://user-images.githubusercontent.com/79643476/147557276-4251f952-f3db-4960-8cc5-33b4846ad162.png)

Possibly the conv1_bn/cond is actually no supported by OpenVINO.
We'll further investigate this and get back to you soon.

**Comment by avitial** [comment]
@JinXiaozhao based on error messages thrown by the MO and glancing the model in Netron, there are multiple ops of type `If` which currently are not supported in 2021.4.2 release of OpenVINO. The first op of this type MO encounters is in _conv1_bn/cond_, so the error is expected. 

Also it looks like support for `IF` operation was added by [PR #6296](https://github.com/openvinotoolkit/openvino/pull/6296) and merged to master branch, I've tried converting the frozen model of resnet50_ckpt.py to IR unfortunately without success. 

In addition glancing at the repo files from the provided URL to resnet50_ckpt.py script, I came across [resnet50_tensorflow.py](https://github.com/Adlik/Adlik/blob/master/benchmark/tests/test_model/resnet50_tensorflow/resnet50_tensorflow.py). The resulting frozen model from this script seems to convert just fine to IR, wonder if this is useful to you.

```
$ python3 /opt/intel/openvino_2021/deployment_tools/model_optimizer/mo.py --input_model frozen_graph.pb --batch 1
Model Optimizer arguments:
Common parameters:
        - Path to the Input Model:      /home/user/cases/git/9452-I_get_problem_when_i_convert_checkpoint_model_to_openvino/model/frozen_graph.pb
        - Path for generated IR:        /home/user/cases/git/9452-I_get_problem_when_i_convert_checkpoint_model_to_openvino/model/.
        - IR output name:       frozen_graph
        - Log level:    ERROR
        - Batch:        1
        - Input layers:         Not specified, inherited from the model
        - Output layers:        Not specified, inherited from the model
        - Input shapes:         Not specified, inherited from the model
        - Mean values:  Not specified
        - Scale values:         Not specified
        - Scale factor:         Not specified
        - Precision of IR:      FP32
        - Enable fusing:        True
        - Enable grouped convolutions fusing:   True
        - Move mean values to preprocess section:       None
        - Reverse input channels:       False
TensorFlow specific parameters:
        - Input model in text protobuf format:  False
        - Path to model dump for TensorBoard:   None
        - List of shared libraries with TensorFlow custom layers implementation:        None
        - Update the configuration file with input/output node names:   None
        - Use configuration file used to generate the model with Object Detection API:  None
        - Use the config file:  None
        - Inference Engine found in:    /opt/intel/openvino_2021/python/python3.8/openvino
Inference Engine version:       2021.4.2-3974-e2a469a3450-releases/2021/4
Model Optimizer version:        2021.4.2-3974-e2a469a3450-releases/2021/4
2021-12-30 16:25:00.499535: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/intel/openvino_2021/data_processing/dl_streamer/lib:/opt/intel/openvino_2021/data_processing/gstreamer/lib:/opt/intel/openvino_2021/opencv/lib:/opt/intel/openvino_2021/deployment_tools/ngraph/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/tbb/lib::/opt/intel/openvino_2021/deployment_tools/inference_engine/external/hddl/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/omp/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/gna/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/mkltiny_lnx/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/lib/intel64
2021-12-30 16:25:00.499580: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
/home/user/.local/lib/python3.8/site-packages/tensorflow/python/autograph/impl/api.py:22: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
[ WARNING ]
Detected not satisfied dependencies:
        tensorflow: installed: 2.5.2, required: ~= 2.4.1

Please install required versions of components or use install_prerequisites script
/opt/intel/openvino_2021.4.752/deployment_tools/model_optimizer/install_prerequisites/install_prerequisites_tf2.sh
Note that install_prerequisites scripts may install additional components.
[ SUCCESS ] Generated IR version 10 model.
[ SUCCESS ] XML file: /home/user/cases/git/9452-I_get_problem_when_i_convert_checkpoint_model_to_openvino/model/frozen_graph.xml
[ SUCCESS ] BIN file: /home/user/cases/git/9452-I_get_problem_when_i_convert_checkpoint_model_to_openvino/model/frozen_graph.bin
[ SUCCESS ] Total execution time: 17.83 seconds.
[ SUCCESS ] Memory consumed: 910 MB.
```

**Comment by jgespino** [comment]
Closing, please re-open if additional assistance is needed.

**Comment by JinXiaozhao** [comment]
Sorry for only replying now, is this the problem of the operator conv1_bn/cond ? @Iffa-Meah 
Thanks for your help, but we still want to use this model，https://github.com/Adlik/Adlik/blob/master/benchmark/tests/test_model/resnet50_tensorflow/resnet50_ckpt.py

**Comment by avitial** [comment]
@JinXiaozhao apologies for the delay in my response, as previously stated it looks like the problem comes from operator conv1_bn/cond. I was not successful at creating the model, freezing and converting to IR with TensorFlow 2.x and OpenVINO 2021.4.2 LTS. I however was successful with TensorFlow 1.15 and OpenVINO 2021.4.2 LTS.

If possible I'd suggest trying with TensorFlow 1.15 and see if this alternative method works for you. Note in OpenVINO 2021.4 TensorFlow 1.x moved to supported with limitations, refer to the [Release Notes](https://www.intel.com/content/www/us/en/developer/articles/release-notes/openvino-relnotes.html) for additional information.

Best Regards,
Luis

**Comment by avitial** [comment]
Closing this, please re-open if needed.
